### PR TITLE
BW-1206 - Combine all Wes Endpoints & add Tests

### PR DESCRIPTION
**Pull request by kpierre13**

### CODE REVIEW

File: engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala [hunk 1]
```diff
@@ -149,4 +154,53 @@ class WesRouteSupportSpec extends AsyncFlatSpec with ScalatestRouteTest with Mat
...
+          status
+        }
+        headers should be(Seq.empty)
+      }
+  }
+
+
+  behavior of "WES API /runs GET endpoint"
+  it should "return results for a good query" in {
+    Get(s"/ga4gh/wes/v1/runs?status=Succeeded&id=${CromwellApiServiceSpec.ExistingWorkflowId}") ~>
```
Comment (salonishah11) [inline engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala:183]: Can you remind me which API this is for? I don't see GET `/runs` endpoint accepting `status` and `workflowId` parameters

**Comment by salonishah11** [comment]
@kpierre13 these tests cases look great. Would it also make sense to add tests for below endpoints? Or do tests for these already exist?
- GET `/runs/{workflowId}/status`
- POST `/runs/{workflowId}/cancel`
- GET `/service-info`

**Comment by kpierre13** [comment]
> @kpierre13 these tests cases look great. Would it also make sense to add tests for below endpoints? Or do tests for these already exist?
> 
> * GET `/runs/{workflowId}/status`
> * POST `/runs/{workflowId}/cancel`
> * GET `/service-info`

There are existing tests for the first two in the same file. The tests for `service-info` are located in [ServiceInfo.spec](https://github.com/broadinstitute/cromwell/blob/21fc23f41b776d666a089b7c83b64a066bc730e8/engine/src/test/scala/cromwell/webservice/routes/wes/ServiceInfoSpec.scala)

File: engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala (hunk 1 above)
Comment (kpierre13) [reply engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala:183]: These are for the Wes Api for the GET /runs endpoint. The endpoint returns an array that contains elements with "run_id" and "state", so my idea here was to check that we are getting back those two parameters from the query.

Comment (salonishah11) [reply engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala:183]: Right. But I don't think GET `/runs` accepts `status` and `id` as query parameters no? Here the request should be this?
```
Get("/ga4gh/wes/v1/runs") ~>
```

Comment (kpierre13) [reply engine/src/test/scala/cromwell/webservice/routes/wes/WesRouteSupportSpec.scala:183]: I see. Refactoring now...

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala [hunk 2]
```diff
@@ -49,36 +56,53 @@ trait WesRouteSupport extends HttpInstrumentation {
...
                 }
-              )
+            },
+            path("runs" / Segment) { workflowId =>
+              get {
+                // this is what it was like in code found in the project… it perhaps isn’t ideal but doesn’t seem to hurt, so leaving it like this for now.
+                completeCromwellResponse(runLog(workflowId, (w: WorkflowId) => GetSingleWorkflowMetadataAction(w, None, None, expandSubWorkflows = false), serviceRegistryActor))
+              }
+            },
+              path("runs" / Segment / "status") { possibleWorkflowId =>
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala:82]: Indentation seems a little off here?

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala [hunk 3]
```diff
@@ -49,36 +56,53 @@ trait WesRouteSupport extends HttpInstrumentation {
         pathPrefix("ga4gh" / "wes" / "v1") {
           concat(
             path("service-info") {
-              complete(ServiceInfo.toWesResponse(workflowStoreActor))
+              get {
+                complete(ServiceInfo.toWesResponse(workflowStoreActor))
+              }
             },
-            pathPrefix("runs") {
```
Comment (cjllanwarne) [inline engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala:54]: I see you've removed this path prefix and added a `"runs"` to the start of all the cases. That's fine, just curious whether it was a deliberate choice?

**Review by cjllanwarne — APPROVED** [review]

**Review by salonishah11 — APPROVED** [review]

File: engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala (hunk 3 above)
Comment (aednichols) [reply engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala:54]: +1 on curiosity, I do think `pathPrefix` is more conventional to reduce repetition, but I'm sure this works fine too

Comment (kpierre13) [reply engine/src/main/scala/cromwell/webservice/routes/wes/WesRouteSupport.scala:54]: I originally intended to use `pathPrefix` for repetitions sake, but doing so confused all of the Routes for some reason, and the fix I found was to simply give all the routes their own `runs`.

**Pull request merged**